### PR TITLE
updated artefact DTOs

### PR DESCRIPTION
**Pull request by Bronson-Brown-deVost**
@Itay here are the updates to the ArtefactDTO system you wanted.  Let me know if I missed something.

### CODE REVIEW

File: sqe-api-server/Serialization/ArtefactConversion.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        {
+            return new ArtefactDTO
+            {
+                id = artefact.ArtefactId,
+                editionId = editionId,
+                imageId = artefact.ImageId,
+                artefactDataEditorId = artefact.ArtefactDataEditorId,
+                mask = artefact.Mask,
+                artefactMaskEditorId = artefact.MaskEditorId,
+                isPlaced = artefact.Rotate.HasValue,
```
Comment (zmbq) [inline sqe-api-server/Serialization/ArtefactConversion.cs]: I think `isPlaced` should be true if the translation is defined. We can have a rotation even if the artefact hasn't been placed (if we rotate it in the artefact editor)

File: sqe-dto/Transformation.cs [hunk 2]
```diff
@@ -3,24 +3,33 @@
...
         public uint zIndex { get; set; }
+
+        [Required]
         public TranslateDTO translate { get; set; }
     }
 
     public class TranslateDTO
     {
-        [Required] public uint x { get; set; }
+        [Required]
```
Comment (zmbq) [inline sqe-dto/Transformation.cs]: A required nullable?

Comment (Bronson-Brown-deVost) [reply sqe-dto/Transformation.cs]: Again, you are right.  That is a mistake.  Thanks.

File: sqe-api-server/Serialization/ArtefactConversion.cs (hunk 1 above)
Comment (Bronson-Brown-deVost) [reply sqe-api-server/Serialization/ArtefactConversion.cs]: Oops! you are right.

**Review by zmbq — APPROVED** [review]

**Pull request merged**